herbert_core/classes/MPIFramework/@MPI_clusters_factory/private/select_and_set_working_parallel_cluster_.m

```diff
@@ -47,23 +47,24 @@ function select_and_set_working_parallel_cluster_(obj,val)
             numel(names),val);
     end
     cl_name = names{val};
-    
+
 else
     error('HERBERT:MPI_clusters_factory:invalid_argument',...
         'Unknown parallel cluster: ''%s'' type: ''%s'' requested',...
         strtrim(evalc('disp(val)')),class(val));
 end
-obj.parallel_cluster_name_ = cl_name;
 cl = obj.known_clusters_(cl_name);
+obj.parallel_cluster_name_ = cl_name;
 try
     cl.check_availability();
+    obj.framework_available_   = true;
 catch ME
+    obj.framework_available_   = false;
     if strcmp(ME.identifier,'HERBERT:ClusterWrapper:not_available')
-        obj.framework_available_ = false;
         return
     else
         rethrow(ME);
     end
 end
-obj.framework_available_ = true;
+
```

Comment:
Is this form common enough we want to make it a function?

Reply:
do not know -- I do not remember it wider usage.